src/server/tests/test_handlers.py

```diff
@@ -0,0 +1,171 @@
+        self.saved_user_doc_action_tuple = (user, document, action_on_doc)
+
+
+class HandlersTests(unittest.TestCase):
+
+    def setUp(self):
+        self.app = main.APP.test_client()
+        main.APP.config['TESTING'] = True
+        self.dal = DalMock()
+        handlers.DAL = self.dal
+
+    def test_home_without_user_render_login(self):
+        response = self.app.get('/', follow_redirects=True)
+        self._assert_is_login(response)
+
+    def test_home_with_user_render_docs(self):
+        self._login()
+        response = self.app.get('/', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_login_with_user_connected_redirect_home(self):
+        self._login()
+        response = self.app.get('/login', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_login_post_valid_redirect_home(self):
+        response = self.app.post('/login', data=dict(email='mark', password='dadada'), follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_login_post_invalid_redirect_login(self):
+        response = self.app.post('/login', data=dict(email='mark', password='JoIsNoOne'), follow_redirects=True)
+        self._assert_is_login(response)
+        self.assertTrue('Unknown email or invalid password' in response.data)
+
+    def test_disconnect_disconnect_and_redirect_login(self):
+        self._login()
+        response = self.app.get('/disconnect', follow_redirects=True)
+        with self.app.session_transaction() as session:
+            self.assertFalse('email' in session)
+        self._assert_is_login(response)
+
+    def test_disconnect_not_connected_redirect_login(self):
+        response = self.app.get('/disconnect', follow_redirects=True)
+        self._assert_is_login(response)
+
+    def test_register_get_without_user_display_register(self):
+        response = self.app.get('/register', follow_redirects=True)
+        self._assert_is_register(response)
+
+    def test_register_get_with_user_redirect_home(self):
+        self._login()
+        response = self.app.get('/register', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_register_post_with_user_redirect_home(self):
+        self._login()
+        response = self.app.post('/register', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_register_post_with_existing_email(self):
+        response = self.app.post('/register', data=dict(email='mark', password='', interests=''), follow_redirects=True)
+        self._assert_is_register(response)
+        self.assertTrue('This account already exists' in response.data)
+
+    def test_register_post_with_new_email_save_profile_redirect_login(self):
+        post_data = dict(email='alan', password='alan', interests='rockets\r\ncars')
+        response = self.app.post('/register', data=post_data, follow_redirects=True)
+        self.dal.saved_password = pswd.hash_password('alan')
```

Comment:
What is the pupose of this line ? I suppose that the initial desire was to write an assert ?